pkg/auth/dependency/authnsession/provider_impl.go

```diff
@@ -155,7 +155,7 @@ func (p *providerImpl) NewFromScratch(userID string, principalID string, reason
 	}, nil
 }
 
-func (p *providerImpl) GenerateResponse(authnSess auth.AuthnSession) (interface{}, error) {
+func (p *providerImpl) GenerateResponseAndUpdateLastLoginAt(authnSess auth.AuthnSession) (interface{}, error) {
 	step, ok := authnSess.NextStep()
 	if !ok {
 		var authInfo authinfo.AuthInfo
```

Comment:
I think this entire block should be refactored into `SessionProvider`: it is logically creating a session with needed extra step. Maybe fold it into `SessionProvider.Create`? Then, we can keep the same name, since creating a session should always update `last_login_at`.

Reply:
That would greatly increase the dependency of SessionProvider.

Reply:
Okay let's just keep it as is for now.